example-apps/chatbot-rag-app/data/index_data.py

```diff
@@ -82,20 +110,86 @@ def main():
 
     print(f"Split {len(workplace_docs)} documents into {len(docs)} chunks")
 
-    print(f"Creating Elasticsearch sparse vector store in {ELASTICSEARCH_URL}")
+    print(f"Creating Elasticsearch sparse vector store for {ELASTICSEARCH_URL}")
 
-    elasticsearch_client.indices.delete(index=INDEX, ignore_unavailable=True)
-
-    ElasticsearchStore.from_documents(
-        docs,
-        es_connection=elasticsearch_client,
+    store = ElasticsearchStore(
+        es_connection=es,
         index_name=INDEX,
         strategy=ElasticsearchStore.SparseVectorRetrievalStrategy(model_id=ELSER_MODEL),
-        bulk_kwargs={
-            "request_timeout": 60,
-        },
     )
 
+    # The first call creates ML tasks to support the index, and typically fails
+    # with the default 10-second timeout, at least when Elasticsearch is a
+    # container running on Apple Silicon.
+    #
+    # Once elastic/elasticsearch#107077 is fixed, we can use bulk_kwargs to
+    # adjust the timeout.
+
+    print(f"Adding documents to index {INDEX}")
+
+    spinner = Halo(text="Processing bulk operation", spinner="dots")
```

Comment:
Being Python where there's no block scope, you could put this inside the `if` if you want

Reply:
I did and my IDE complains about it, which is why I didn't ;)

Reply:
<img width="290" alt="Screenshot 2025-02-20 at 9 14 56 AM" src="https://github.com/user-attachments/assets/c6945fad-ccec-4e08-b40d-d3265a4c8387" />

Reply:
testing now and I expect you are right that even though IDE doesn't like this, it is probably fine in python